### PR TITLE
Add useZeroed partner to use

### PR DESCRIPTION
**Pull request by kortschak**
The choice of a separate function rather than a bool in use is highlight when the returned slice is expected to be zeroed; a true is likely to be ignored when reading.

Also added test of zeroing and moved zero into matrix.go.

### CODE REVIEW

File: mat64/matrix.go [hunk 1]
```diff
@@ -430,3 +430,23 @@ func use(f []float64, l int) []float64 {
...
+func useZeroed(f []float64, l int) []float64 {
+	if l <= cap(f) {
+		f = f[:l]
+		zero(f)
+		return f
+	}
+	return make([]float64, l)
+}
+
+// zero does a fast zeroing of the given slice's elements.
```
Comment (btracey) [inline mat64/matrix.go]: This might actually be slower in 1.5  https://groups.google.com/forum/#!topic/golang-codereviews/lx3hsZr4HZE

Comment (btracey) [inline mat64/matrix.go]: Sorry, this comment is in reference to the for loop.

Comment (kortschak) [inline mat64/matrix.go]: That's fine. It's a single location to change when the time comes.

**Comment by btracey** [comment]
LGTM
